src/inspect.cpp

```diff
@@ -123,6 +123,8 @@ namespace Sass {
     if (dec->value()->concrete_type() == Expression::NULL_VAL) return;
     bool was_decl = in_declaration;
     in_declaration = true;
+    if (dec->is_custom_property()) in_custom_property = true;
```

Comment:
Maybe use LOCAL_FLAG macro to ensure it is reset automatically when scope is left (e.g. due to thrown error).

Reply:
Updated